py-polars/tests/unit/io/test_scan.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+import pytest
+
+import polars as pl
+from polars.testing.asserts.frame import assert_frame_equal
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from polars.type_aliases import SchemaDict
+
+
+def _scan(file_path: Path, schema: SchemaDict | None = None) -> pl.LazyFrame:
+    suffix = file_path.suffix
+    if suffix == ".ipc":
+        return pl.scan_ipc(file_path)
+    if suffix == ".parquet":
+        return pl.scan_parquet(file_path)
+    if suffix == ".csv":
+        return pl.scan_csv(file_path, schema=schema)
+    msg = f"Unknown suffix {suffix}"
+    raise NotImplementedError(msg)
```

Comment:
Does polars already provide a uniform `scan` interface?

Reply:
This doesn't exist yet, at least not on the Python side.